analyses/project-updates/01-generate-project-report.Rmd

```diff
@@ -0,0 +1,420 @@
+
+cellranger_results_file <- file.path(cellranger_results_dir, "QC_Summary_CellRanger_Report.tsv")
+SoupX_results_file <- file.path(SoupX_results_dir, "SoupX_contamination_summary.tsv")
+scDblFinder_results_file <- file.path(scDblFinder_results_dir, "doublets_metrics.tsv")
+cells_table_file <- file.path(Filter_object_results_dir, "number_of_cells_per_sample.tsv")
+#metadata_file <- file.path(metadata_dir, "project_metadata.tsv") # metadata input file
+
+summary_samples_table_file <- file.path(metadata_dir, metadata_file) # summary_samples_table file
+```
+
+
+# Methods
+
+## Data
+
+### `r PROJECT_NAME`
+
+```{r read-table, echo = FALSE}
+summary_samples_table <- readr::read_tsv(summary_samples_table_file, guess_max = 100000, show_col_types = FALSE)
+```
+
+```{r, fig.align = "left", results = "asis", message = FALSE, warning = FALSE, echo = FALSE}
+# Create the data frame
+tables1 <- summary_samples_table
+cat("  \n<div align=\"center\" style=\"font-size:80%\">  \n")
+print(knitr::kable(tables1, align = "lcccc", caption = "Summary of samples in the cohort"))
+cat("  \n</div>  \n")
+cat("  \n\\pagebreak  \n")
+```
+
+  
+# Results
+
+All results can be found here:
+
+- Cluster users can access data from: `r root_dir`/analyses/
+- Mac users can access data from: `r mac_results_filepath`/`r cohort_value`/analyses/
+- `r results_filepath`/`r cohort_value`/analyses/
+
+
+We used FastQC (v0.11.9) and multiqc (v1.25) to perform quality control tool for all libraries across the project from the raw sequencing reads. Next, we aligned these reads using CellRanger (v8.0.1) (10x Flex Genomics). 
+
+## fastqc-analysis
+
+FastQC metrics such as per-base sequence quality, GC content, and adapter contamination were within expected thresholds, ensuring that the data is suitable for downstream analyses. Further validation steps, such as alignment or feature counting, will help confirm that the data is suitable for further processing and analysis.
+
+For more information, please see: `r results_filepath`/`r cohort_value`/analyses/fastqc-analysis/results
+
+### Results: `r PROJECT_NAME` cohort
+
+```{r read-fastqc-analysis, echo = FALSE}
+#multiqc_general_stats_df <- read.table(multiqc_general_stats_results_file, header = TRUE, sep = "\t") %>%
+#  arrange(Sample)
+
+#multiqc_fastqc_df <- read.table(multiqc_fastqc_results_file, header = TRUE, sep = "\t") %>% 
+#  mutate_if(is.numeric, ~ scales::comma(.)) %>%
+#  arrange(Sample)
+
+fastqc_overrepresented_df <- read.table(fastqc_overrepresented_results_file, header = TRUE, sep = "\t") %>%
+  rename_with(~ gsub(" ", "_", .)) %>% # Replace spaces with underscores in column names
+  mutate(Overrepresented_sequences_by_sample_pct = Top.overrepresented.sequence + Sum.of.remaining.overrepresented.sequences) %>%
+  mutate_if(is.numeric, ~ round(. , 2)) %>% # Round all numeric columns to 2 decimal places
+  arrange(Sample)
+```
+
+#### Sequence Quality Histograms
+- The results are within the expected range.
```

Comment:
This may not always be a true statement, so I think it could be more informative if this section briefly explained what patterns for this metric might indicate an issue with the sequencing data and what some common explanations for seeing an issue might be, like you do in the sections below.

I think a couple sentences sharing this type of info could be summarized based off the fastqc documentation [here](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/3%20Per%20Sequence%20Quality%20Scores.html). E.g. this graph indicates the overall quality scores for the sequences. If a large proportion of sequences have lower scores this can indicated a general quality issue with the sequencing run.